server/controllers/finance/vouchers.js

```diff
@@ -328,23 +326,32 @@ function createVoucher(voucherDetails, userId, projectId) {
     item = db.convert(item, ['uuid', 'voucher_uuid', 'document_uuid', 'entity_uuid']);
   });
 
-  // map items into an array of arrays
-  items = _.map(
-    items,
-    util.take('uuid', 'account_id', 'debit', 'credit', 'voucher_uuid', 'document_uuid', 'entity_uuid'),
-  );
-
   // initialise the transaction handler
   const transaction = db.transaction();
 
   // build the SQL query
-  transaction
-    .addQuery('INSERT INTO voucher SET ?', [voucherDetails])
-    .addQuery(
-      'INSERT INTO voucher_item (uuid, account_id, debit, credit, voucher_uuid, document_uuid, entity_uuid) VALUES ?',
-      [items],
-    )
-    .addQuery('CALL PostVoucher(?);', [voucherDetails.uuid]);
+  transaction.addQuery('INSERT INTO voucher SET ?', [voucherDetails]);
+
+  items.forEach(item => {
+    // lookup for cost_center_id from account_id if not given
+    transaction.addQuery(
+      `INSERT INTO voucher_item (
+        uuid, account_id, debit, credit, voucher_uuid, document_uuid, entity_uuid, cost_center_id
+      ) VALUES (?, ?, ?, ?, ?, ?, ?, ${item.cost_center_id ? '?' : 'GetCostCenterByAccountId(?)'})`,
```

Comment:
Can we just do this in SQL entirely?
```sql
IFNULL(?, GetCostCenterByAccountId(?))
```
It might be cleaner than having `?` in Javascript.

Reply:
The thing is, we have to deal with `cost_center_id` and `account_id` for this column, when the `cost_center_id` is not defined we use the `account_id` else we use the given `cost_center_id`.

Using `IFNULL` imply that if the variable given is `null` use the function `GetCostCenterByAccountId` on this `null` variable